src/inmanta/compiler/__init__.py

```diff
@@ -205,7 +214,35 @@ def compile(self):
         statements.append(requires_rel)
         return (statements, blocks)
 
+    def export_data(self) -> None:
+        """
+            Exports compiler data if the option has been set.
+        """
+
+        def do_write(f: TextIO) -> None:
+            # wrap between start and end markers if writing to stdout
+            wrap: bool = f == sys.stdout
+            if wrap:
+                f.write("---START export-compile-data---\n")
```

Comment:
When testing this out, it turns out that, when running the compiler as follows:

```
inmanta compile --export-compile-data 2>/dev/null
```

The markers are not required, i.e. there is only json.

Perhaps it would beter to not have the markers (sorry, I suggested them), so that you can cleanly pipe the output to e.g. jq.

Reply:
There are two reasons I opted to follow your suggestion and use the markers:
- using `-v` also prints to stdout. You could argue that you shouldn't use `-v` when you want structured output though.
- There are a lot of `print` statements throughout the code. I wasn't sufficiently confident that none of them would trigger in any compile run. This is the main reason. If you are sufficiently confident, then the markers could indeed be removed.

Reply:
@bartv what do you think is the best way to get clean json out?

Reply:
I do not see why output to stdout is required. Additionally will this code fail if export() at some point prints log lines. I propose that this is only written to a file. From experience we know that this markers work well until you hit that one corner case :)

Reply:
ok, let's drop the std,... (sorry)

Reply:
No problem, the suggestion made sense